x/crosschain/keeper/grpc_query_get_tss_address.go

```diff
@@ -21,10 +21,23 @@ func (k Keeper) GetTssAddress(goCtx context.Context, req *types.QueryGetTssAddre
 	}
 
 	ctx := sdk.UnwrapSDKContext(goCtx)
-
-	tss, found := k.GetTSS(ctx)
+	var tss types.TSS
+	found := false
+	tssList := k.GetAllTSS(ctx)
+	if req.TssPubKey == "" {
+		tss, found = k.GetTSS(ctx)
+		if !found {
+			return nil, status.Error(codes.NotFound, "current tss not set")
+		}
+	}
 	if !found {
-		return nil, status.Error(codes.NotFound, "not found")
+		for _, t := range tssList {
+			if t.TssPubkey == req.TssPubKey {
+				tss = t
+				found = true
```

Comment:
The `found` is set to true and used nowhere afterwards. Maybe we can do this

```
	ctx := sdk.UnwrapSDKContext(goCtx)
	var tssPubKey string
	if req.TssPubKey == "" {
		tss, found := k.GetTSS(ctx)
		if !found {
			return nil, status.Error(codes.NotFound, "current tss not set")
		}
		tssPubKey = tss.TssPubkey
	} else {
		tssList := k.GetAllTSS(ctx)
		for _, t := range tssList {
			if t.TssPubkey == req.TssPubKey {
				tssPubKey = t.TssPubkey
				break
			}
		}
	}
	ethAddress, err := getTssAddrEVM(tssPubKey)
	if err != nil {
		return nil, status.Error(codes.Internal, err.Error())
	}
	btcAddress, err := getTssAddrBTC(tssPubKey)
	if err != nil {
		return nil, status.Error(codes.Internal, err.Error())
	}
```

Reply:
Found does the same thing as the else condition, But I do agree this looks more readable